src/collector/directory_entries.rs

```diff
@@ -0,0 +1,88 @@
+use std::fs::{self, DirEntry};
+use std::io::ErrorKind;
+use std::path::PathBuf;
+
+use crate::{report::ReportEntry, Collector, CrateInfo, Result};
+
+use super::CollectionError;
+
+/// List information about entries in a directory.
+///
+/// Limitations:
+/// * Is not recursive
+/// * Does not handle symbolic links
+/// * Only sizes of files are printed and not e.g. time of last modification
+///
+/// # Example
+///
+/// ```md
+/// #### File and dir
+///
+/// - file.txt, 14 bytes
+/// - some_dir
+///
+/// ```
+pub struct DirectoryEntries {
+    title: String,
+    path: PathBuf,
+}
+
+impl DirectoryEntries {
+    pub fn new(title: impl Into<String>, path: impl Into<PathBuf>) -> Self {
+        Self {
+            title: title.into(),
+            path: path.into(),
+        }
+    }
+}
+
+impl Collector for DirectoryEntries {
+    fn description(&self) -> &str {
+        &self.title
+    }
+
+    fn collect(&mut self, _: &CrateInfo) -> Result<ReportEntry> {
+        let path_str = &self.path.to_string_lossy();
+
+        let mut entries = fs::read_dir(&self.path)
+            .map_err(|e| read_dir_error_to_report_entry(e, path_str))?
+            .map(|e| match e {
+                Ok(dir_entry) => dir_entry_to_report_entry(dir_entry),
+                Err(e) => format!("Error: {}", e),
+            })
+            .collect::<Vec<_>>();
+
+        // For stable ordering
+        entries.sort();
+
+        if entries.is_empty() {
+            Ok(ReportEntry::Text(format!("'{}' is empty", path_str)))
+        } else {
+            Ok(ReportEntry::List(
+                entries.into_iter().map(ReportEntry::Text).collect(),
+            ))
+        }
+    }
+}
+
+fn read_dir_error_to_report_entry(error: std::io::Error, path_str: &str) -> CollectionError {
+    CollectionError::CouldNotRetrieve(if error.kind() == ErrorKind::NotFound {
+        format!("'{}' not found", path_str)
+    } else {
+        format!("'{}' not read: {}", path_str, error)
+    })
+}
+
+fn dir_entry_to_report_entry(dir_entry: DirEntry) -> String {
+    let mut text = String::new();
+
+    text.push_str(&dir_entry.file_name().to_string_lossy());
+
+    if let Ok(metadata) = dir_entry.metadata() {
+        if metadata.is_file() {
+            text.push_str(&format!(", {} bytes", metadata.len()));
+        }
+    }
```

Comment:
Could we maybe add a trailing `/` or `\` for directories (https://doc.rust-lang.org/std/path/constant.MAIN_SEPARATOR.html) to make them easier to distinguish from files/symlinks?

Also, I saw that you mentioned that there is no special handling for symlinks. But calling `metadata` instead of `symlink_metadata` will imply that we also show the file size of the linked files. We could maybe do a minimal symlink "handling" here by using [`symlink_metadata`](https://doc.rust-lang.org/std/fs/fn.symlink_metadata.html) and then append `@` like in `ls --classify`?

```
       -F, --classify
              append indicator (one of */=>@|) to entries
```

Reply:
Great idea about suffixing dir with `MAIN_SEPARATOR`! Will do.

Also seems like adding appending `@` for symlinks and using `symlink_metadata()` instead is straightforward. I will take a shot at it.

Reply:
Fix pushed. I just realised I might also need to adjust the test case on Windows to expect `\` instead of `/`, but that will be easy and I will wait and see if CI actually fails first.

Reply:
Oh no, https://doc.rust-lang.org/stable/std/fs/struct.Metadata.html#method.is_symlink is only available in Rust 1.58. It is very nice to have because then we don't to mess around with `symlink_metdata()`, as demonstrated by that my automated tests passes with Rust 1.58 but that does not use `symlink_metadata()`.

Are you OK with us descoping symlink support again? I'd rather not have to mess around with symlinks :(

Reply:
Ok, sounds good to me. Thanks for looking into it.